rust/private/rust_analyzer.bzl

```diff
@@ -72,20 +73,16 @@ def write_rust_analyzer_spec_file(ctx, attrs, owner, base_info):
 
     return rust_analyzer_info
 
-def _accumulate_rust_analyzer_info(dep_infos_to_accumulate, label_index_to_accumulate, dep):
+def _accumulate_rust_analyzer_info(deps_details, dep):
```

Comment:
nit: (this is somewhat pre-existing, but the new version makes it a bit worse): on a first read, I don't see what deps_details looks like. It becomes clear when I reach the initialization below. 

Maybe could you declare a provider for the shape of this and use it in docs.

Reply:
Yeah, starlark doesn't have useful types :-(

Providers do indeed have the required structure but I think using them for non-provider data (while calculating provider data) is going to be actively misleading.

There's not much logic here, and I think it's easier to just inline it so the data structures can be understood from context.